### PR TITLE
CCData bugfix

### PR DESCRIPTION
**Pull request by seobyeongky**
Hi
I fixed a Dangerous bug.

### CODE REVIEW

File: cocos/base/CCData.cpp [hunk 1]
```diff
@@ -39,13 +39,17 @@
...
     move(other);
 }
 
-Data::Data(const Data& other)
+Data::Data(const Data& other) :
+_bytes(nullptr),
+_size(0)
 {
     CCLOGINFO("In the copy constructor of Data.");
     copy(other._bytes, other._size);
```
Comment (dumganhar) [inline cocos/base/CCData.cpp:55]: Why is it dangerous? 
In `Data::move` and `Data::copy`, the member variables will be assigned.

Comment (dumganhar) [inline cocos/base/CCData.cpp:55]: But I think your fix is all right.

Comment (seobyeongky) [inline cocos/base/CCData.cpp:55]: Ah, Because 'copy' calls 'clear'. and 'clear' calls 'free(_bytes)'. _byte has unexpected value. so dangerous.

Comment (dumganhar) [inline cocos/base/CCData.cpp:55]: Ouch, it's my fault. :) Thanks

Comment (seobyeongky) [inline cocos/base/CCData.cpp:55]: Welcome

**Pull request merged**